.github/workflows/lint.yml

```diff
@@ -21,21 +31,12 @@ jobs:
     strategy:
       fail-fast: false
     runs-on: ubuntu-latest
-    env:
-      GOPROXY: https://proxy.golang.org/,direct
-      GOPRIVATE: ""
-      GONOPROXY: ""
-      GONOSUMDB: github.com/github/*
     steps:
-      - uses: actions/setup-go@v5
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
         with:
-          go-version: ${{ vars.GOVERSION }}
-          check-latest: true
-      - uses: actions/checkout@v4
+          go-version-file: 'go.mod'
       - name: Lint
-        # This also does checkout, setup-go, and proxy setup.
-        uses: github/go-linter@v1.2.1
-        with:
-          strict: true
-          go-version: ${{ vars.GOVERSION }}
-          goproxy-token: ${{secrets.GOPROXY_TOKEN}}
+        uses: golangci/golangci-lint-action@971e284b6050e8a5849b72094c50ab08da042db8
```

Comment:
Using a different linter since the other was giving errors about "Error: Unable to resolve action `github/go-linter`, not found".